File: patch_dynamic.py
for file_path in ['src/app/login/page.tsx', 'src/app/connect/page.tsx']:
    with open(file_path, 'r') as f:
        content = f.read()
    
    if "export const dynamic = 'force-dynamic';" not in content:
        # insert after imports
        parts = content.split('\n\n', 1)
        if len(parts) > 1:
            content = parts[0] + "\n\nexport const dynamic = 'force-dynamic';\n\n" + parts[1]
            
        with open(file_path, 'w') as f:
            f.write(content)

File: patch_dynamic.js
const fs = require('fs');

const files = [
  'src/app/snapshot/[id]/page.tsx',
  'src/app/snapshot/[id]/compare/page.tsx',
  'src/app/story/[id]/page.tsx',
  'src/app/journey/[id]/page.tsx'
];

const dynamicStr = `
export const dynamic = 'force-dynamic';
export const revalidate = 0;
export const fetchCache = 'force-no-store';
`;

for (const file of files) {
  if (fs.existsSync(file)) {
    let code = fs.readFileSync(file, 'utf8');
    if (!code.includes('export const dynamic')) {
      // Insert after imports
      const lastImportIndex = code.lastIndexOf('import ');
      const endOfImport = code.indexOf('\n', lastImportIndex);
      code = code.slice(0, endOfImport + 1) + dynamicStr + code.slice(endOfImport + 1);
      fs.writeFileSync(file, code);
      console.log(`Patched ${file}`);
    }
  }
}
